g2p/mappings/utils.py

```diff
@@ -803,30 +803,34 @@ def validate_preserve_case(self):
             )
         return self
 
-    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
-    # Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
-    @validator("rules_path", "abbreviations_path", "alignments_path", pre=True)
-    def add_parent_dir(cls, path, values):
-        if isinstance(path, str):
-            path = Path(path)
-        # Append the parent directory to the path
-        if isinstance(path, Path) and "parent_dir" in values and values["parent_dir"]:
-            path = Path(values["parent_dir"]) / path
-        return path
-
-    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
-    # Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
-    @validator("language_name", always=True)
-    def create_language_name(cls, language_name, values):
-        # Default language name to in_lang
-        if language_name is None:
-            language_name = values["in_lang"]
-        return language_name
-
-    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
-    # Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
-    @validator("display_name", always=True)
-    def create_display_name(cls, display_name, values):
-        if display_name is None:
-            display_name = f"{values['in_lang']} {find_mapping_type(values['in_lang'])} to {values['out_lang']} {find_mapping_type(values['out_lang'])}"
-        return display_name
+    @field_validator(
+        "rules_path", "abbreviations_path", "alignments_path", mode="before"
+    )
+    @classmethod
+    def add_parent_dir(cls, value: Any, info: ValidationInfo):
+        """If there is a parent directory, prepend it to all path fields."""
+        parent_dir = info.data.get("parent_dir", None)
+        if value is not None and parent_dir:
```

Comment:
since value is type Any here, it might be good to check `isinstance(value, string)` or something, maybe a try/except here? If someone passes an invalid type to any of the fields validated by this validator it will fail without a proper `pydantic.ValidationError` since the mode is `before`

Reply:
That's quite disappointing. The field is still typed  `Optional[Path]`, so I expect that validation to still happen at some point during the processing,

But OK, fixing it.